src/app/shared/map/map.component.ts

```diff
@@ -108,6 +120,50 @@ export class MapComponent implements OnInit, OnDestroy {
 
   }
 
+
+  ngAfterViewInit(): void {
+    if (this.map && this.map.googleMap) {
+      this.directionsRenderer.setMap(this.map.googleMap);
+    } else {
+      console.error('Mapa nie jest gotowa, sprawdź inicjalizację mapy.');
```

Comment:
eng pls

Reply:
nice to have, not blocking:

again the same situation with if-else, and logging errors instead of handling them:

```
    if (!this.map && !this.map.googleMap) {
      console.error('Mapa nie jest gotowa, sprawdź inicjalizację mapy.');
      return;
    } 

      this.directionsRenderer.setMap(this.map.googleMap);
```